x-pack/legacy/plugins/xpack_main/public/hacks/check_xpack_info_change.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { identity } from 'lodash';
+import { uiModules } from 'ui/modules';
+import { Path } from 'plugins/xpack_main/services/path';
+import { xpackInfo } from 'plugins/xpack_main/services/xpack_info';
+import { xpackInfoSignature } from 'plugins/xpack_main/services/xpack_info_signature';
+
+const module = uiModules.get('xpack_main', []);
+
+module.factory('checkXPackInfoChange', ($q, Private, $injector) => {
+  /**
+   *  Intercept each network response to look for the kbn-xpack-sig header.
+   *  When that header is detected, compare its value with the value cached
+   *  in the browser storage. When the value is new, call `xpackInfo.refresh()`
+   *  so that it will pull down the latest x-pack info
+   *
+   *  @param  {object} response - the angular $http response object
+   *  @param  {function} handleResponse - callback, expects to receive the response
+   *  @return
+   */
+  function interceptor(response, handleResponse) {
+    if (Path.isUnauthenticated()) {
+      return handleResponse(response);
+    }
+
+    const currentSignature = response.headers('kbn-xpack-sig');
+    const cachedSignature = xpackInfoSignature.get();
+
+    if (currentSignature && cachedSignature !== currentSignature) {
```

Comment:
That was removed in https://github.com/elastic/kibana/pull/51818 by mistake.
Xpack cannot detect license update without it. I also noticed that it doesn't work well with different fetching mechanism. Some plugins use NP `fetch`, some custom fetching mechanism, for example https://github.com/restrry/kibana/blob/354a152caa94d856b8ffe1fa4df0981421c1e134/x-pack/legacy/plugins/canvas/common/lib/fetch.ts#L10
https://github.com/restrry/kibana/blob/354a152caa94d856b8ffe1fa4df0981421c1e134/x-pack/legacy/plugins/infra/public/pages/metrics/containers/with_metrics.tsx#L6-L8
We need to talk to solution teams about unifying the network stack.
@elastic/kibana-platform

Reply:
It seems we need to solve several problems:
- to unify the network stack whenever possible
- to investigate and document how we can integrate http interceptors in any network library. Apollo supports
middlewares and afterwares https://www.apollographql.com/docs/react/networking/network-layer/#afterware That could require adjusting interceptor API.